### PR TITLE
Fix Inconsistent behaviour on db:prepare task

### LINKED ISSUE #36383: Inconsistent behavior on db:prepare task
The `db:prepare` task works different for PG adapter than `SQLite3` one.

`db:prepare` task introduced in https://github.com/rails/rails/pull/35768 in case of missing database is supposed to create it, load schema and seeds.

This is working fine on PostgreSQL database, however when using SQLite3 database, seeds are not loaded.
This is happening, because `db:migrate` does not raise `ActiveRecord::NoDatabaseError` which `db:prepare` relies on https://github.com/rails/rails/blob/eece0bf1087dc2cb59605a5a9beb88c6b935c194/activerecord/lib/active_record/railties/databases.rake#L272

Not sure what the proper fix would be. Raising `ActiveRecord::NoDatabaseError` on SQLite3 would be backward incompatible change, so maybe we should add some internal method for existence check of db, that could be used instead of relying on raising error?
/cc @robertomiranda

### PR DESCRIPTION
**Pull request by robertomiranda**
## Summary

Now that `database_exists?`  has been added to connection adapters #36471 we can use this method in order to prevent inconsistencies described on #36383. Basically, SQLite never raise an `ActiveRecord::NoDatabaseError` exception since it creates the files or stores the information on memory every time that tries to establish a connection 

closes #36383 

cc @guilleiguaran @eileencodes

### CODE REVIEW

File: activerecord/lib/active_record/railties/databases.rake [hunk 1]
```diff
@@ -295,24 +295,24 @@ db_namespace = namespace :db do
     seed = false
 
     ActiveRecord::Base.configurations.configs_for(env_name: ActiveRecord::Tasks::DatabaseTasks.env).each do |db_config|
-      ActiveRecord::Base.establish_connection(db_config.config)
+      if ActiveRecord::Base.database_exists?(db_config)
+        ActiveRecord::Tasks::DatabaseTasks.migrate
 
-      ActiveRecord::Tasks::DatabaseTasks.migrate
+        # Skipped when no database
```
Comment (morgoth) [inline activerecord/lib/active_record/railties/databases.rake]: I guess this comment can now be deleted. Previously there was an error happening line above, so the flow was not obvious by looking at the code, but now it is quite straightforward.

Comment (guilleiguaran) [reply activerecord/lib/active_record/railties/databases.rake]: 👍

**Comment by robertomiranda** [comment]
Update: `ActiveRecord::Base.database_exists?` was not added as part of #36471, so I have added this method as part of the Base class as well.

**Comment by morgoth** [comment]
It should be enough to rebase with master to have this method included.
There are also failed specs - can you look into it?

**Comment by robertomiranda** [comment]
> It should be enough to rebase with master to have this method included.

I've rebased with master `database_exists?` has been added at the adapters level #36471, so this PR also adds `database_exists` to `ActiveRecord::Base.database_exists?`

> There are also failed specs - can you look into it?

they are green now ✅

File: activerecord/lib/active_record/railties/databases.rake [hunk 2]
```diff
@@ -295,24 +295,23 @@ db_namespace = namespace :db do
...
-        nil,
-        db_config.env_name,
-        db_config.spec_name
-      )
+      if ActiveRecord::Base.database_exists?(db_config.config)
+        ActiveRecord::Tasks::DatabaseTasks.migrate
+        ActiveRecord::Tasks::DatabaseTasks.dump_schema(db_config.config, ActiveRecord::Base.schema_format, db_config.spec_name)
+      else
+        ActiveRecord::Tasks::DatabaseTasks.create_current(db_config.env_name, db_config.spec_name)
+        ActiveRecord::Tasks::DatabaseTasks.dump_schema(db_config.config, ActiveRecord::Base.schema_format, db_config.spec_name)
```
Comment (morgoth) [inline activerecord/lib/active_record/railties/databases.rake]: @robertomiranda Why did you add dumping schema in this place? There is actually nothing to dump as the database is created step above

**Comment by fatkodima** [comment]
@robertomiranda Are you interested in finishing this PR?
